### PR TITLE
MJ Implement cashbox autobill service ACE01-476 

### PR DESCRIPTION
**Pull request by maryjoes2004**
Updated cashbox service to add subscription items (products) to existing subscription

### CODE REVIEW

File: lib/cashbox/model/subscription.rb [hunk 1]
```diff
@@ -47,28 +47,19 @@ def grace_period?
...
-    end
-
-    def update_subscription_items
-      Cashbox::Request.new(:post, "#{route(id)}?effective_date=today&bill_prorated_period=true", { body: self.to_json })
+    def add_subscription_items(subscprition_items)
+      request = Cashbox::Request.new(:post, route(vid), 
+          {
+              query: { 
+                  effective_date: 'today',
+                  bill_prorated_period: true
```
Comment (sgelbart) [inline lib/cashbox/model/subscription.rb]: bill_prorated_period works? I thought it didn't work with direct debit?

Comment (maryjoes2004) [reply lib/cashbox/model/subscription.rb]: it does not work for direct debit. I have asked Danny if this should be merged or held till Vindicia can fix it.

File: lib/cashbox/model/subscription.rb [hunk 2]
```diff
@@ -47,28 +47,19 @@ def grace_period?
...
+              },
+              body: {
+                  id: id,
+                  items: subscprition_items
```
[the remark this answers is not in the document]
Comment (maryjoes2004) [reply lib/cashbox/model/subscription.rb]: fixed spelling

File: spec/unit/model/subscription_spec.rb [hunk 3]
```diff
@@ -89,140 +89,45 @@
...
+         subscription.add_subscription_items(subscprition_items, true)
     end
 
     it 'makes the correct api call to cashbox' do
       expect(Cashbox::Request).to have_received(:new)
-        .with(
-          :post, '/subscriptions/sub_1235?effective_date=today&bill_prorated_period=true',
-          { body: subscription.to_json }
-        )
+        .with(:post, '/subscriptions/sub_1235',
```
Comment (maulin) [inline spec/unit/model/subscription_spec.rb]: add a test to test the return value. The code that casts the response and returns it.

Comment (maryjoes2004) [reply spec/unit/model/subscription_spec.rb]: updated test

**Comment by maryjoes2004** [comment]
ACE01-476 Cashbox - add product for existing subscription
Closes #87 
LGTM @danny @maulin @sgelbart

**Pull request merged**